gcovr/utils.py

```diff
@@ -180,7 +180,14 @@ def __init__(self, regex, path_context=None):
 
 class Filter(object):
     def __init__(self, pattern):
-        self.pattern = re.compile(pattern)
+        cwd = os.getcwd()
+        # Gueesing if file system is case insensitive.
+        # The working directory is not the root and accessable in upper and lower case.
+        is_fs_case_insensitive = ( cwd != os.path.sep ) and os.path.exists(cwd.upper()) and os.path.exists(cwd.lower())
+        flags = 0
```

Comment:
Do we need other flags here in the near future? If not this could be changed to:
```
flags = re.IGNORECASE if is_fs_case_insensitive else 0
```

I am not sure if this coding style is ok here.

Reply:
Changed. The `x if y else z` is often used.